src/main/java/com/guysfromusa/carsgame/control/round/SameTypeOnePerCarGameRoundSelector.java

```diff
@@ -0,0 +1,40 @@
+package com.guysfromusa.carsgame.control.round;
+
+import com.guysfromusa.carsgame.control.Command;
+import com.guysfromusa.carsgame.control.MessageType;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+/**
+ * Created by Robert Mycek, 2018-05-05
+ */
+@Component
+public class SameTypeOnePerCarGameRoundSelector implements GameRoundSelector {
+
+    @Override
+    public Optional<GameRound> selectCommand(Queue<Command> queue, String gameName) {
+        if (queue.isEmpty()) {
```

Comment:
it cant be empty at this point, is there a situation when optional will be empty?

Reply:
actually no